R/api_calls.R

```diff
@@ -57,7 +79,93 @@ See ?ed_auth.")
 Contact technology@eidith.org.")
   }
 
-  if(verbose) message("Importing...")
+  if(verbose) cat_line("Importing...\n")
+  data <- fromJSON(content(request, as = "text", encoding="UTF-8"))
+
+  if(header_only) {
+    return(data)
+  } else {
+    data <- as_tibble(data)
+  }
+
+  # if(nrow(data) == 0){
+  #   message((paste0("Download for the ", endpoint, " table failed. Please contact technology@eidith.org")))
+  #   return(invisible(0))
+  # }
+
+  if("ExceptionMessage" %in% names(data)){
+    cat_line(red(paste0("Download for the ", endpoint, " table failed. Please contact technology@eidith.org\n")))
+    return(invisible(0))
+  }
+
+  if(postprocess){
+    data <- tryCatch(ed_process(data, endpoint),
+                     error = function(e){
+                       cat_line(red(paste0("Error: There are unexpected fields in the ", endpoint, " download. Please contact technology@eidith.org.\n")))
+                       return(invisible(NULL))
+                     })
+
+}
+
+  return(data)
+}
+
+
+#' @param verbose Show a progress bar and other messages?
+#' @param header_only Return only the table header.  Useful for checking if
+#' API access works
+#' @param lmdate_from filter records by earliest date, in YYYY-MM-DD format
+#' @param lmdate_to filter records by latest date, in YYYY-MM-DD format (see details)
+#' @param ... additional arguments passed to [httr::GET()]
+#' @return a [tibble][tibble::tibble()]-style data frame
+#' @rdname ed2_get
+#' @name ed2_get
+
```

Comment:
`ed2*` functions don't have a help file.  We should have one that consolidates all the E2 table functions.  You can probably put the `ed_get` and `ed2_get` functions in a single help file.

Reply:
The `ed2*` documentation would be a good place to put a short version or link to a vignette describing what all the tables are.